### PR TITLE
Ds 3563 missing index metadatavalue resource type id

### PR DESCRIPTION
**Pull request by tomdesair**
This is a fix for https://jira.duraspace.org/browse/DS-3563

### CODE REVIEW

File: dspace-api/src/main/resources/org/dspace/storage/rdbms/sqlmigration/oracle/V5.7_2017.04.11__DS-3563_Index_metadatavalue_resource_type_id_column.sql [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+--
+-- http://www.dspace.org/license/
+--
+
+------------------------------------------------------
+-- DS-3563 Missing database index on metadatavalue.resource_type_id
+------------------------------------------------------
+-- Create an index on the metadata value resource_type_id column so that it can be searched efficiently.
+
+CREATE INDEX metadatavalue_resource_type_id_idx ON metadatavalue (resource_type_id);
```
Comment (tdonohue) [inline dspace-api/src/main/resources/org/dspace/storage/rdbms/sqlmigration/oracle/V5.7_2017.04.11__DS-3563_Index_metadatavalue_resource_type_id_column.sql]: The Oracle migration should also either check for existence of the index *or* (like Postgres & H2) drop the index if it exists and recreate it. Here's a few resources for how to do that:
* Check if index exists before creation in Oracle: http://openacs.org/xowiki/cond_create_index
* Drop index if exists in Oracle: https://stackoverflow.com/questions/2722630/oracle-drop-index-if-exists

**Review by tdonohue — CHANGES_REQUESTED** [review]
Looks good overall (other than Oracle comments). But, because this is backporting a database migration (to a previous release), we have to *be very careful* about how it gets applied. As-is, for sites that have installed DSpace 6.0 already, this `V5.7` migration would be ignored (as Flyway will assume it is only for 5.x).

So, it ensure it is applied to 6.0 sites as well, we will need a secondary `V6.1` migration on the `dspace-6_x` branch (in a separate PR) which can safely run whether or not this `V5.7` migration already ran.

The other option is to simply note this performance issue as a bug in 5.x and only fix it in a single `V6.1` migration. But, I realize that may not be ideal for sites that are encountering this on 5.x

**Comment by tomdesair** [comment]
I've added the conditional create for Oracle. 

I don't think we need a DSpace 6.x version of this index because that column doesn't exist in the DSpace 6 schema (since we use UUIDs and a dspaceobject table there).

**Review by tdonohue — APPROVED** [review]

**Review by kshepherd — APPROVED** [review]
All builds passed, unit tests passed, smoke testing against my test repository passed. Since this is a quick win with no 6.x issues and a +1 from Tim already, I'll merge this now.

**Comment by kshepherd** [comment]
Thanks @tomdesair ! A great quick performance win.

**Pull request merged**